zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateEventSubprocessTest.java

```diff
@@ -13,20 +13,22 @@
 import io.camunda.zeebe.engine.util.EngineRule;
 import io.camunda.zeebe.model.bpmn.Bpmn;
 import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessMessageSubscriptionIntent;
 import io.camunda.zeebe.protocol.record.value.BpmnElementType;
 import io.camunda.zeebe.test.util.BrokerClassRuleHelper;
 import io.camunda.zeebe.test.util.record.RecordingExporter;
 import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Map;
 import org.assertj.core.api.Assertions;
-import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TestWatcher;
 
 public class MigrateEventSubprocessTest {
 
-  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+  @Rule public final EngineRule engine = EngineRule.singlePartition();
```

Comment:
Yes, I found the issue. JUnit is using the same engine instance with `@ClassRule` and running test methods in parallel. Since we publish a message in test cases the published message name and correlation key is also in other test cases. I resolved the issue by replacing all correlation keys with `helper.getCorrelationValue()`. 👍